File: src/item/Seed.java
package item;

public interface Seed {
    /**
     * Grow the current Seed.
     */
    void grow();

    /**
     * @return The FoodType of the Seed return after harvest
     */
    FoodType harvest();

    /**
     * @return The state of the seed
     */
    SeedState getSeedState();

    /**
     * @return The grow time
     */
    double getGrowTime();

    /**
     * @return The FoodType related to the Seed
     */
    FoodType getFoodType();

    /**
     * @return The SeedType of the current Seed
     */
    SeedType getSeedType();
}

File: C#/RegaSimone/FarmingSimulator/FarmingSimulator/Seed.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace FarmingSimulator
{
    interface Seed
    {
        void Grow();
        FoodType Harvest();
        SeedState GetSeedState();
        double GetGrowTime();
        FoodType GetFoodType();
        SeedType GetSeedType();

    }
}
